src/deployment/deploy.py

```diff
@@ -694,42 +689,48 @@ def set_instance_config(self) -> None:
             tenants.append(tenant)
         update_allowed_aad_tenants(config_client, tenants)
 
-    def create_eventgrid(self) -> None:
-        logger.info("creating eventgrid subscription")
-        src_resource_id = self.results["deploy"]["fuzz-storage"]["value"]
-        dst_resource_id = self.results["deploy"]["func-storage"]["value"]
-
+    def remove_eventgrid(self) -> None:
         credential = AzureCliCredential()
         client = StorageManagementClient(
             credential, subscription_id=self.get_subscription_id()
         )
-        event_subscription_info = EventSubscription(
-            destination=StorageQueueEventSubscriptionDestination(
-                resource_id=dst_resource_id, queue_name="file-changes"
-            ),
-            filter=EventSubscriptionFilter(
-                included_event_types=[
-                    "Microsoft.Storage.BlobCreated",
-                    "Microsoft.Storage.BlobDeleted",
-                ]
-            ),
-            retry_policy=RetryPolicy(
-                max_delivery_attempts=30,
-                event_time_to_live_in_minutes=1440,
-            ),
-        )
+        try:
+            storage_accounts = client.storage_accounts.list_by_resource_group(
+                self.resource_group
+            )
+            for storage_account in storage_accounts:
+                if storage_account.name.startswith("fuzz"):
+                    src_resource_id = storage_account.id
+
+            if not src_resource_id:
+                return
+        except ResourceNotFoundError:
+            return
 
         client = EventGridManagementClient(
             credential, subscription_id=self.get_subscription_id()
         )
-        result = client.event_subscriptions.begin_create_or_update(
-            src_resource_id, "onefuzz1", event_subscription_info
-        ).result()
-        if result.provisioning_state != "Succeeded":
-            raise Exception(
-                "eventgrid subscription failed: %s"
-                % json.dumps(result.as_dict(), indent=4, sort_keys=True),
-            )
+
+        old_subscription_name = "onefuzz1"
+
+        def old_subscription_exists() -> bool:
+            try:
+                client.event_subscriptions.get(src_resource_id, old_subscription_name)
+                return True
+            except ResourceNotFoundError:
+                return False
+
+        if old_subscription_exists():
+            logger.info("removing deprecated event subscription")
+            result = client.event_subscriptions.begin_delete(
+                src_resource_id, old_subscription_name
+            ).result()
+
+            if result and result.provisioning_state != "Succeeded":
```

Comment:
If `result` is absent / evals `False`, then does that mean something else has gone wrong? Should we have a check like:
```python
if not result:
    raise Exception(...)
```

Reply:
the check on result was actually not necessary. The operation return a polling object.